python/cudf/cudf/_lib/table.pyx

```diff
@@ -114,7 +114,9 @@ cdef class Table:
         for _ in column_names:
             data_columns.append(Column.from_unique_ptr(move(dereference(it))))
             it += 1
-        data = dict(zip(column_names, data_columns))
+        data = ColumnAccessor._create_unsafe(
+            dict(zip(column_names, data_columns))
+        )
```

Comment:
Seems like above we create a list by appending (since we need to bump the iterator). Instead of creating this intermediate list can we go straight to the dictionary as we iterate over the column names?

Reply:
Used a dict comprehension instead and it cleans things up nicely. Also while doing so, I noticed that we can use the `_unsafe` constructor to build the Index as well.